helix_fhir_client_sdk/graph/simulated_graph_processor_mixin.py

```diff
@@ -243,79 +243,83 @@ async def _process_target_async(
-                                id_=child_id,
-                                cache=cache,
-                                scope_parser=scope_parser,
-                                logger=logger,
-                            )
-                            responses.append(child_response)
-                            children = child_response.get_bundle_entries()
-                            if logger:
-                                logger.info(
-                                    f"Received child resources"
-                                    + f" from parent {parent_resource.get('resourceType')}/{parent_resource.get('id')}"
-                                    + f" [{path}]"
-                                    + f", count:{len(child_response.get_resource_type_and_ids())}, cached:{cache_hits}"
-                                    + f", {','.join(child_response.get_resource_type_and_ids())}"
-                                )
-            else:  # single reference
-                if parent_resource and parent_resource.get(path) and target_type:
-                    reference = parent_resource.get(path)
-                    if reference and "reference" in reference:
-                        reference_id = reference["reference"]
-                        reference_parts = reference_id.split("/")
-                        if reference_parts[0] == target_type:
-                            child_id = reference_parts[1]
-                            (
-                                child_response,
-                                cache_hits,
-                            ) = await self._get_resources_by_parameters_async(
-                                session=session,
-                                resource_type=target_type,
-                                id_=child_id,
-                                cache=cache,
-                                scope_parser=scope_parser,
-                                logger=logger,
+
+        # forward link and iterate over list
+        if path and "[x]" in path:
+            references: Union[List[Dict[str, Any]], Dict[str, Any], str, None] = (
+                DictionaryParser.get_nested_property(parent_resource, path)
+                if parent_resource
+                else None
+            )
+            # remove null references
+            if references and isinstance(references, list):
+                references = [r for r in references if r is not None]
+
+            if parent_resource and references and target_type:
+                reference_ids = []
+                for r in references:
+                    # TODO: consider removing assumption of "reference" and require as part of path instead
+                    if isinstance(r, dict) and "reference" in r:
+                        reference_ids.append(r["reference"])
+                    elif isinstance(r, str) and r.startswith("Binary/"):
+                        reference_ids.append(r)
```

Comment:
This is the part I mentioned before that works for this specific issue, but it isn't the best solution.  My preference would be to require the full `path` in the GraphDefinition (e.g. `foo.bar.reference` instead of just `foo.bar`), but that would require changes to all of the current graphs and feels like a bigger project than this bug fix.

Reply:
Agreed -- may make sense to log a ticket/feature request, assuming the "more correct" approach also entails updating all existing `GraphDefinitions` to accommodate

Reply:
i created https://icanbwell.atlassian.net/browse/IOP-2814 for it.  in addition to revisiting the best way to do this, we might be able to use something like jmespath to completely eliminate the need for DictionaryParser altogether